spec/filters/grok_spec.rb

```diff
@@ -702,4 +702,66 @@
     end
   end
 
+  describe  "grok with nil coerced value" do
+    config <<-CONFIG
+      filter {
+        grok {
+          match => { "message" => "test (N/A|%{BASE10NUM:duration:float}ms)" }
+        }
+      }
+    CONFIG
+
+    sample "test 28.4ms" do
+      insist { subject["duration"] } == 28.4
+      insist { subject["tags"] }.nil?
+    end
+
+    sample "test N/A" do
+      insist { insist { subject.to_hash }.include?("duration") }.fails
```

Comment:
As a fun thing that I liked while designing insist, was that you could use `insist { ... }` as the subject:

```
subject do
  insist { something }
end

it "should be empty" do
  subject.empty?
end
```

It's not as explicit as I like, but I kind of like how it works. This pattern can be used with rspec's expect via `subject { expect(object) }` but due to expect's grammar pattern ("expect XYZ to ...") it doesn't read as well.

Some day we'll move the test suite fully away from insist and into using rspec expectations